### PR TITLE
feat: Move all logs everything over to logger

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

We've had a few issues with over-logging. These show up in recordings and are unclear who they belong to as we don't prefix, as well as generally being a lot noisier than people like.

* Everything now goes via `logger` 
* console.log is no longer allowed via eslint
* Only `critical` logs are always logged
* Otherwise debug need to be enabled or a new window value `POSTHOG_DEBUG` to be present (helps us enable debugging on people's sites where `posthog` is not available in the global context

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers

### CODE REVIEW

File: src/gdpr-utils.ts [hunk 1]
```diff
@@ -230,41 +230,7 @@ export function userOptedOut(posthog: PostHog, silenceErrors: boolean | undefine
...
-/**
- * Wrap a method with a check for whether the user is opted out of data capturing and cookies/localstorage for the given token
- * If the user has opted out, return early instead of executing the method.
- * If a callback argument was provided, execute it passing the 0 error code.
- * @param {PostHog} posthog - the posthog instance
- * @param {function} method - wrapped method to be executed if the user has not opted out
- * @param silenceErrors
- * @returns {*} the result of executing method OR undefined if the user has opted out
- */
-export function addOptOutCheck<M extends (...args: any[]) => any = (...args: any[]) => any>(
```
Comment (benjackwhite) [inline src/gdpr-utils.ts:249]: This was never called. Guessing it was some legacy thing that never got removed

File: src/__tests__/decide.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ describe('Decide', () => {
     given('decide', () => new Decide(given.posthog))
     given('posthog', () => ({
         config: given.config,
+        debug: true,
```
Comment (pauldambra) [inline src/__tests__/decide.js:28]: is this needed given we set the Window value too?

**Review by pauldambra — APPROVED** [review]
LGTM

File: src/__tests__/decide.js (hunk 2 above)
Comment (benjackwhite) [reply src/__tests__/decide.js:28]: Good point

**Review by mariusandra — APPROVED** [review]

**Pull request merged**